content/5_Plat/Sweep_Line.mdx

```diff
@@ -71,13 +71,36 @@ TC 760 ComponentsForever -->
 
 ## Radial Sweep
 
-<IncompleteSection />
+Instead of a vertical line sweeping the plane from left to right, **radial
+sweep** involves a ray that rotates around a central point (like a radar
+screen):
+
+![Radar screen gif](https://i.gifer.com/7jnn.gif)
+
+In this case, we sort points/events by their
+[bearing](<https://en.wikipedia.org/wiki/Bearing_(angle)>) instead of by their
+x- and y-coordinates. Besides that, the mechanics are the same as those of
+normal line sweep.
 
 <FocusProblem problem="radialSam" />
 
 ### Solution - Seeing the Boundary
 
-<IncompleteSection />
+**Complexity:** $\mathcal O(N + Rp_i \log R)$
+
+In this problem, there are three types of events: when our ray hits a fence
+post, enters a rock, or exits a rock.
+
+The second and third types of events can be found for each rock by sorting the
+rays to its vertices by bearing and then taking the two endpoints of the sorted
+list. These two rays are the two tangents to the rock.
+
+We can then perform a radial sweep to find the fence posts that Farmer Don can
+see - these fence posts are simply the ones where the number of type-2 and
+type-3 events we've processed so far are equal.
+
+Note that some optimizations (e.g. inserting fence posts in a specific order to
+avoid sorting) may be required to get 100 points.
 
 <LanguageSection>
```

Comment:
I don't think you need `atan2`?

Reply:
What should I use? I'm not sure whether sorting by cross product is the same as sorting by bearing

Reply:
then why do you do both?

Reply:
I don't know how to compare bearings using cross product without a bunch of casework, so I feel like using `atan2` is simpler in this case. I compare by cross product when finding the tangents to the rocks because sorting by bearing would be incorrect if the rock crossed the y-axis

Reply:
see https://old.yosupo.jp/submission/13604

Reply:
Should I not explicitly construct the list of fence posts to achieve that?